pint/models/frequency_dependent.py

```diff
@@ -0,0 +1,61 @@
+"""This module implements a frequency evolution of pulsar profiles model"""
+from warnings import warn
+import parameter as p
+from .timing_model import TimingModel, Cache
+import astropy.units as u
+import numpy as np
+import pint.utils as ut
+import astropy.time as time
+
+class FD(TimingModel):
+    """This class provides a timing model for frequency evolution of pulsar
+    profiles model.
+    """
+    def __init__(self):
+        super(FD, self).__init__()
+        self.add_param(p.prefixParameter(name='FD1', units="second", value=0.0,
+                       descriptionTplt=lambda x: ("%d term of frequency"
+                                                  " dependent  coefficients" % x),
+                       unitTplt=lambda x: 'second',
+                       type_match='float'))
+
+        self.delay_funcs['L1'] += [self.FD_delay]
```

Comment:
I agree with you that 'L1' and 'L2' are not the perfect name. 'SolarSystem'  and 'PulsarSystem' seems hard to apply to dispersion delay.